src/plugins/vis_builder/public/application/utils/use/use_persisted_agg_params.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { integer } from '@opensearch-project/opensearch/api/types';
+import { AggGroupNames, CreateAggConfigParams } from '../../../../../data/common';
+import { Schema } from '../../../../../vis_default_editor/public';
+
+export const usePersistedAggParams = (
+  types,
+  oldAggParams: CreateAggConfigParams[],
+  oldVisType?: string,
+  newVisType?: string
+): CreateAggConfigParams[] => {
+  if (oldVisType && newVisType) {
+    const oldVisualizationType = types.get(oldVisType)?.ui.containerConfig.data.schemas.all;
+    const newVisualizationType = types.get(newVisType)?.ui.containerConfig.data.schemas.all;
+    const aggMapping = getSchemaMapping(oldVisualizationType, newVisualizationType);
+    const newAggParams = oldAggParams.map((newAggParam: CreateAggConfigParams) =>
```

Comment:
the old/new naming here feels confusing, like how can an element of the `oldAggParams` array be a `newAggParam`? Id just call `oldAggParams` as `aggConfigParams`
```suggestion
    const updatedAggConfigParams = aggConfigParams.map((aggConfigParam: CreateAggConfigParams) =>
```